StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1515CodeFixProvider.cs

```diff
@@ -77,5 +78,79 @@ private static SyntaxNode ProcessTriviaList(SyntaxNode syntaxRoot, SyntaxTriviaL
 
             return syntaxRoot.ReplaceTrivia(triviaList[0], new[] { SyntaxFactory.CarriageReturnLineFeed, triviaList[0] });
         }
+
+        private class FixAll : DocumentBasedFixAllProvider
+        {
+            public static FixAllProvider Instance { get; } =
+                new FixAll();
+
+            protected override string CodeActionTitle =>
+                LayoutResources.SA1515CodeFix;
+
+            protected override async Task<SyntaxNode> FixAllInDocumentAsync(FixAllContext fixAllContext, Document document)
+            {
+                var diagnostics = await fixAllContext.GetDocumentDiagnosticsAsync(document).ConfigureAwait(false);
+                var syntaxRoot = await document.GetSyntaxRootAsync().ConfigureAwait(false);
+
+                List<SyntaxTrivia> trivias = new List<SyntaxTrivia>();
+
+                foreach (var diagnostic in diagnostics)
+                {
+                    trivias.Add(syntaxRoot.FindTrivia(diagnostic.Location.SourceSpan.Start));
+                }
+
+                var tokensWithTrivia = trivias.GroupBy(x => x.Token);
+
+                Dictionary<SyntaxToken, SyntaxToken> replacements = new Dictionary<SyntaxToken, SyntaxToken>();
+
+                foreach (var tokenWithTrivia in tokensWithTrivia)
+                {
+                    var token = tokenWithTrivia.Key;
+                    var newLeadingTrivia = FixTriviaList(token.LeadingTrivia, tokenWithTrivia);
+                    var newTrailingTrivia = FixTriviaList(token.TrailingTrivia, tokenWithTrivia);
+
+                    replacements.Add(token, token.WithLeadingTrivia(newLeadingTrivia).WithTrailingTrivia(newTrailingTrivia));
+                }
+
+                return syntaxRoot.ReplaceTokens(replacements.Select(x => x.Key), (oldToken, newToken) => replacements[oldToken]);
+            }
+
+            private static SyntaxTriviaList FixTriviaList(SyntaxTriviaList triviaList, IEnumerable<SyntaxTrivia> commentTrivias)
```

Comment:
:question: Do we gain anything (simplicity) by taking this method and `ProcessTriviaList` and creating one method that both the fix all and single fix implementations can use?